Week3/homework/step2-6.js

```diff
@@ -1,16 +1,18 @@
 'use strict';
 
+// Test PASS
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
+// Adham: Array.prototype.flat() is not accepted here.
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return arr.reduce((firstLeft, nextRight) => firstLeft.concat(nextRight));
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const firstLevel = arr.reduce((firstLeft, nextRight) => firstLeft.concat(nextRight));
+  const secondLevel = firstLevel.reduce((firstLeft, nextRight) => firstLeft.concat(nextRight));
+  return secondLevel;
```

Comment:
Very well. 👍 

If you would like to think about it further, try and search on google how you could implement it using `recursion`.